.werft/platform-delete-preview-environments-cron.ts

```diff
@@ -80,12 +85,64 @@ class HarvesterPreviewEnvironment {
         ])
     }
 
-    isInactive(): boolean {
-        // We'll port over the logic from CoreDevPreviewEnvironment later, for now we consider
-        // Harvester preview environments to never be stale due to inactivity.
-        const sliceID = SLICES.CHECKING_FOR_NO_DB_ACTIVITY
-        werft.log(sliceID, `${this.name} (${this.namespace}) - is-inactive=false - Harvester based `)
-        return false
+    /**
+     * Checks whether a preview environment is active based on the db activity.
+     *
+     * It errs on the side of caution, so in case of connection issues etc. it will consider the
+     * preview environment active.
+     */
+    isActive(): boolean {
+        const sliceID = SLICES.CHECKING_FOR_DB_ACTIVITY
+        try {
+            const vm = VM.get({name: this.name })
+            const vmErrNotFound = `Error from server (NotFound): virtualmachineinstances.kubevirt.io "${this.name}" not found`
+            if (vm.code != 0){
+                if (vm.stderr == vmErrNotFound){
```

Comment:
Checking directly on the stderr seems a bit fragile to me. How about having VM.get return a specific `VMNotFound` error and then check on that here instead? I've left comments on the implementation of VM.geg below on how to construct the error, but here the usage would be something like

```ts
try {
  ....
catch (e) {
  if (e instsanceof VMNotFoundError) {
    // log and return false
  }
  // log and return true
}
```